### PR TITLE
Update `vimeo/psalm` to v5.20.0

### LINKED ISSUE #277: Update `vimeo/psalm` to v5.16 once available
### Feature Request

|    Q        |   A
|------------ | ------
| QA | yes

#### Summary

Once available, we should update to latest `vimeo/psalm` and add FQCN return type checks as in https://github.com/laminas/laminas-serializer/pull/57/files#diff-7b802ded8d384fcaa263497e01da6ac542d253fc894b2e4058442b9ad8f5cfeeR33-R41

### PR DESCRIPTION
**Pull request by boesing**
|    Q          |   A
|-------------- | ------
 QA            | yes

### Description

Bump psalm to latest version.

### CODE REVIEW

File: psalm-baseline.xml [hunk 1]
```diff
@@ -228,11 +223,66 @@
...
+      <code>$args</code>
+      <code>$args</code>
+    </InvalidArgument>
+  </file>
   <file src="src/Storage/Adapter/AbstractAdapter.php">
     <ArgumentTypeCoercion>
       <code>$plugin</code>
     </ArgumentTypeCoercion>
     <InvalidArgument>
+      <code>$args</code>
```
Comment (boesing) [inline psalm-baseline.xml:241]: https://github.com/vimeo/psalm/issues/10579

File: psalm-baseline.xml [hunk 2]
```diff
@@ -363,6 +413,9 @@
     <DocblockTypeContradiction>
       <code>is_array($options)</code>
     </DocblockTypeContradiction>
+    <InvalidArgument>
```
Comment (boesing) [inline psalm-baseline.xml:416]: https://github.com/vimeo/psalm/issues/10579

File: src/Pattern/CaptureCache.php [hunk 3]
```diff
@@ -325,12 +325,6 @@ protected function createDirectoryStructure($pathname)
                 $err = ErrorHandler::stop();
                 throw new Exception\RuntimeException("mkdir('{$pathname}', 0{$oct}, true) failed", 0, $err);
             }
-
-            if ($perm !== false && ! chmod($pathname, $perm)) {
```
Comment (boesing) [inline src/Pattern/CaptureCache.php:329]: I've spotted this one some time ago already (also with psalm AFAIR).

This check is within an if condition `if ($perm === false)`, so `$perm !== false` is never true here.

**Review by gsteel — APPROVED** [review]
LGTM!

**Pull request merged**